core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider.go

```diff
@@ -384,7 +384,8 @@ func (p *logEventProvider) readLogs(ctx context.Context, latest int64, filters [
 		if configUpdateBlock := int64(filter.configUpdateBlock); start < configUpdateBlock {
 			start = configUpdateBlock
 		}
-		logs, err := p.poller.LogsWithSigs(start, latest, filter.topics, common.BytesToAddress(filter.addr), pg.WithParentCtx(ctx))
+		// query logs based on contract address, event sig, and blocks
+		logs, err := p.poller.LogsWithSigs(start, latest, []common.Hash{filter.topics[0]}, common.BytesToAddress(filter.addr), pg.WithParentCtx(ctx))
```

Comment:
could you also update recoverer to use the content filtering after it fetches logs from lp